lib/rule-tester/rule-tester.js

```diff
@@ -272,11 +272,49 @@ class RuleTester {
         this.rules[name] = rule;
     }
 
+    /**
```

Comment:
I'd like to move the type definitions to outside of the class. And it's better if descriptions exist for each property. For example, we did so in [/lib/cli-engine/cli-engine.js](https://github.com/eslint/eslint/blob/master/lib/cli-engine/cli-engine.js).